SU2_CFD/src/solvers/CTransAFTSolver.cpp

```diff
@@ -0,0 +1,598 @@
+      Inlet_TurbVars[iMarker](iVertex,0) = AF_Inf;
+      Inlet_TurbVars[iMarker](iVertex,1) = lnIntermittency_Inf;
+    }
+  }
+
+   const su2double CFL = config->GetCFL(MGLevel)*config->GetCFLRedCoeff_Turb();
+  for (iPoint = 0; iPoint < nPoint; iPoint++) {
+    nodes->SetLocalCFL(iPoint, CFL);
+    nodes->SetAuxVar(iPoint, 0, geometry->nodes->GetWall_Distance(iPoint));
+  }
+  Min_CFL_Local = CFL;
+  Max_CFL_Local = CFL;
+  Avg_CFL_Local = CFL;
+
+  /*--- Add the solver name. ---*/
+  SolverName = "AFT model";
+
+}
+
+void CTransAFTSolver::Preprocessing(CGeometry *geometry, CSolver **solver_container, CConfig *config,
+         unsigned short iMesh, unsigned short iRKStep, unsigned short RunTime_EqSystem, bool Output) {
+  SU2_OMP_SAFE_GLOBAL_ACCESS(config->SetGlobalParam(config->GetKind_Solver(), RunTime_EqSystem);)
+
+  /*--- Upwind second order reconstruction and gradients ---*/
+  CommonPreprocessing(geometry, config, Output);
+}
+
+void CTransAFTSolver::Postprocessing(CGeometry *geometry, CSolver **solver_container, CConfig *config, unsigned short iMesh) {
+
+  /*--- Compute LM model gradients. ---*/
+
+  if (config->GetKind_Gradient_Method() == GREEN_GAUSS) {
+    SetSolution_Gradient_GG(geometry, config, -1);
+  }
+  if (config->GetKind_Gradient_Method() == WEIGHTED_LEAST_SQUARES) {
+    SetSolution_Gradient_LS(geometry, config, -1);
+  }
+
+  AD::StartNoSharedReading();
+  auto* flowNodes = su2staticcast_p<CFlowVariable*>(solver_container[FLOW_SOL]->GetNodes());
+  auto* turbNodes = su2staticcast_p<CTurbVariable*>(solver_container[TURB_SOL]->GetNodes());
+
+  SU2_OMP_FOR_STAT(omp_chunk_size)
+  for (unsigned long iPoint = 0; iPoint < nPoint; iPoint ++) {
+
+    // Here the nodes already have the new solution, thus I have to compute everything from scratch
+    const su2double AF = nodes->GetSolution(iPoint,0);
+    const su2double lnIntermittency = nodes->GetSolution(iPoint,1);
+    const su2double Critical_N_Factor = config->GetN_Critical();
+    /* Ampification Factor term */
+    const su2double vel_u = flowNodes->GetVelocity(iPoint, 0);
+    const su2double vel_v = flowNodes->GetVelocity(iPoint, 1);
+    const su2double vel_w = (nDim == 3) ? flowNodes->GetVelocity(iPoint, 2) : 0.0;
+    const su2double Velocity_Mag = sqrt(vel_u * vel_u + vel_v * vel_v + vel_w * vel_w);
+    const su2double dist_i = geometry->nodes->GetWall_Distance(iPoint);
+    const su2double StrainMag_i = flowNodes->GetStrainMag(iPoint);
+    const su2double Eddy_Viscosity_i = turbNodes->GetmuT(iPoint);
+    const su2double Laminar_Viscosity_i = flowNodes->GetLaminarViscosity(iPoint);
+    const su2double Density_i = flowNodes->GetDensity(iPoint);
+    const su2double c_1 = 100.0;
+    const su2double c_2 = 0.06;
+    const su2double c_3 = 50.0;
+    su2double Temp3 = flowNodes->GetVelocity(iPoint, 0) * nodes->GetAuxVarGradient(iPoint, 0, 0);
+    su2double Temp4 = flowNodes->GetVelocity(iPoint, 1) * nodes->GetAuxVarGradient(iPoint, 0, 1);
```

Comment:
Add some documentation please, if the initial aux var grad is 0 then you set the aux var to 0, then its gradient is 0 and this remains 0 forever? Looks like a bug.